src/ValidAnagram.java

```diff
@@ -0,0 +1,35 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ValidAnagram {
+
+    public static void main(String[] args) {
+        System.out.println(isAnagram("aaba" , "aabb"));
+    }
+
+    public static boolean isAnagram(String s, String t) {
+        if (s.length() != t.length()) return false;
+        Map<Character, Integer> sMap = new HashMap<>();
+        for (char c : s.toCharArray()) {
```

Comment:
For loop from line 14 and for loop from line can be reduced to one loop and containsKey can be used with direct evaluation. Let's see what I mean,

```
for (int i=0; i < s.length(); i++) {
sMap.put(s.charAt(i), sMap.getOrDefault(s.charAt(i), 0) + 1);
sMap.put(t.charAt(i), sMap.getOrDefault(t.charAt(i), 0) - 1);
}

for (char key : sMap.keySet()) {
            if (sMap.get(key) != 0) return false;
}

```